Sources/NIOCore/EventLoop.swift

```diff
@@ -429,6 +440,15 @@ extension EventLoop {
     }
 }
 
+extension EventLoop {
+    #if swift(>=5.9)
+    @available(macOS 14.0, iOS 17.0, watchOS 10.0, tvOS 17.0, *)
+    public var executor: any SerialExecutor {
+        NIODefaultSerialEventLoopExecutor(self)
+    }
```

Comment:
I am wondering if we really should default to this or are better off with a `fatalError`. We are making sure our ELs implement this sensible but if somebody created their own EL and we use the default I am a bit concerned that users will just run into crashes for which we could provide a better `fatalError` message.

Reply:
Yeah probably it's okay since all our ELs do the right thing and there aren't many other ELs out there.

Reply:
Seems okey to leave this IMHO 👍